core/src/avm1/globals/sound.rs

```diff
@@ -206,23 +206,27 @@ impl<'gc> Sound<'gc> {
 }
 
 const PROTO_DECLS: &[Declaration] = declare_properties! {
-    "attachSound" => method(attach_sound; DONT_ENUM | DONT_DELETE | READ_ONLY);
-    "duration" => property(duration; DONT_ENUM | DONT_DELETE | READ_ONLY);
-    "getDuration" => method(duration; DONT_ENUM | DONT_DELETE | READ_ONLY);
-    "setDuration" => method(set_duration; DONT_ENUM | DONT_DELETE | READ_ONLY);
-    "id3" => method(id3; DONT_ENUM | DONT_DELETE | READ_ONLY);
-    "getBytesLoaded" => method(get_bytes_loaded; DONT_ENUM | DONT_DELETE | READ_ONLY);
-    "getBytesTotal" => method(get_bytes_total; DONT_ENUM | DONT_DELETE | READ_ONLY);
     "getPan" => method(get_pan; DONT_ENUM | DONT_DELETE | READ_ONLY);
     "getTransform" => method(get_transform; DONT_ENUM | DONT_DELETE | READ_ONLY);
     "getVolume" => method(get_volume; DONT_ENUM | DONT_DELETE | READ_ONLY);
-    "loadSound" => method(load_sound; DONT_ENUM | DONT_DELETE | READ_ONLY);
-    "position" => property(position; DONT_ENUM | DONT_DELETE | READ_ONLY);
     "setPan" => method(set_pan; DONT_ENUM | DONT_DELETE | READ_ONLY);
     "setTransform" => method(set_transform; DONT_ENUM | DONT_DELETE | READ_ONLY);
     "setVolume" => method(set_volume; DONT_ENUM | DONT_DELETE | READ_ONLY);
-    "start" => method(start; DONT_ENUM | DONT_DELETE | READ_ONLY);
     "stop" => method(stop; DONT_ENUM | DONT_DELETE | READ_ONLY);
+    "attachSound" => method(attach_sound; DONT_ENUM | DONT_DELETE | READ_ONLY);
+    "start" => method(start; DONT_ENUM | DONT_DELETE | READ_ONLY);
+    "getDuration" => method(duration; DONT_ENUM | DONT_DELETE | READ_ONLY | VERSION_6);
+    "setDuration" => method(set_duration; DONT_ENUM | DONT_DELETE | READ_ONLY | VERSION_6);
+    "getPosition" => method(get_position; DONT_ENUM | DONT_DELETE | READ_ONLY | VERSION_6);
+    "setPosition" => method(set_position; DONT_ENUM | DONT_DELETE | READ_ONLY | VERSION_6);
+    "loadSound" => method(load_sound; DONT_ENUM | DONT_DELETE | READ_ONLY | VERSION_6);
+    "getBytesLoaded" => method(get_bytes_loaded; DONT_ENUM | DONT_DELETE | READ_ONLY | VERSION_6);
```

Comment:
Does this imply the swf version check in the function is no longer relevant and could be removed?
```rust
fn get_bytes_loaded<'gc>(
    activation: &mut Activation<'_, 'gc>,
    _this: Object<'gc>,
    _args: &[Value<'gc>],
) -> Result<Value<'gc>, Error<'gc>> {
    if activation.swf_version() >= 6 {
        avm1_stub!(activation, "Sound", "getBytesLoaded");
        Ok(1.into())
    } else {
        Ok(Value::Undefined)
    }
}
```
(same for `getBytesTotal`)

Reply:
I think that's right! I'll remove it